docs/concepts/overview/extending.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Extending your Kubernetes Cluster
+approvers:
+- erictune
+- lavalamp
+- cheftako
+- chenopis
+---
+
+{% capture overview %}
+
+Kubernetes has many ways to configure and extend it.  As a result,
+there is rarely a need to fork or submit patches to the Kubernetes
+project code.  
+
+This guide describes the options for customizing a kubernetes
+cluster.  It is aimed at **Cluster Administrators** who want to
+understand how to adapt their kubernetes cluster to the needs of
+their work environment.  Developers who are prospective **Extension
+Authors** and **Kubernetes Project Contributors** will also find it
+useful as an introduction to what extension points and patterns
+exist, and their tradeoffs and limitations.
+
+{% endcapture %}
+
+
+{% capture body %}
+
+## Overview
+
+Customization approaches can be broadly divided into **configuration**, which only involves changing flags, local configuration files, or API resources; and **extensions**, which involve running additional programs or services.  This document is primarily about extensions.
+
+## Configuration
+
+**Configuration files** and **flags** are documented in the Reference section of the online documentation, under each binary: [kubelet](/docs/admin/kubelet/), [kube-apiserver](/docs/admin/kube-apiserver/),  [kube-controller-manager](/docs/admin/kube-controller-manager/), and [kube-scheduler](/docs/admin/kube-scheduler/).  Flags and configuration files may not always be changable in a hosted Kubernetes service or a distribution with managed installation.  When they are changable, they are usually only changeable by the cluster administrator.    Also, they are subject to change in future Kubernetes versions, and setting them may require restarting processes.  For those reasons, they should be used only when there are no other options.
+
+**Built-in Policy APIs**, such as [ResourceQuota](/docs/concepts/policy/resource-quotas/), [PodSecurityPolicies](/docs/concepts/policy/pod-security-policy/), [NetworkPolicy](/docs/concepts/services-networking/network-policies/) and Role-based Access Control ([RBAC](/docs/admin/authorization/rbac/)), are built-in Kubernetes APIs.  APIs are typically used with hosted Kubernetes services, and with managed Kubernetes installations.   They are declarative and use the same conventions are other Kubernetes resources like pods, so new cluster configuration can be repeatable and be managed the same way as applications.  And, where they are stable, they enjoy a [defined support policy](/docs/reference/deprecation-policy/) like other Kubernetes APIs.  For these reasons, they are preferred over **configuration files** and **flags** where suitable.
```

Comment:
Same here.  Bullet list will make for easier future diff's and review of modifications to the doc.

Reply:
I think I prefer not to use bulleted list for short lists of examples.  They break of the paragraph structure and thus the flow.

Reply:
I agree.  It's a trade off between language flow and future maintenance.

Today with the paragraph a single line of text, any future add/remove/modify of an item in the list results in a git level change of the entire paragraph.  Were it list form you'd get a more simple diff

```
-  old short line
+ new short line
```

which is much more reviewable.